internal/config/config.go

```diff
@@ -25,71 +25,15 @@ import (
 
 	"github.com/mikioh/ipaddr"
 	"github.com/pkg/errors"
-	yaml "gopkg.in/yaml.v2"
+	metallbv1beta1 "go.universe.tf/metallb/api/v1beta1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
 )
 
-// configFile is the configuration as parsed out of the ConfigMap,
-// without validation or useful high level types.
-type configFile struct {
-	Peers          []peer
-	BGPCommunities map[string]string `yaml:"bgp-communities"`
-	Pools          []addressPool     `yaml:"address-pools"`
-	BFDProfiles    []bfdProfile      `yaml:"bfd-profiles"`
-}
-
-type peer struct {
-	MyASN         uint32         `yaml:"my-asn"`
-	ASN           uint32         `yaml:"peer-asn"`
-	Addr          string         `yaml:"peer-address"`
-	SrcAddr       string         `yaml:"source-address"`
-	Port          uint16         `yaml:"peer-port"`
-	HoldTime      string         `yaml:"hold-time"`
-	KeepaliveTime string         `yaml:"keepalive-time"`
-	RouterID      string         `yaml:"router-id"`
-	NodeSelectors []nodeSelector `yaml:"node-selectors"`
-	Password      string         `yaml:"password"`
-	BFDProfile    string         `yaml:"bfd-profile"`
-	EBGPMultiHop  bool           `yaml:"ebgp-multihop"`
-}
-
-type nodeSelector struct {
-	MatchLabels      map[string]string      `yaml:"match-labels"`
-	MatchExpressions []selectorRequirements `yaml:"match-expressions"`
-}
-
-type selectorRequirements struct {
-	Key      string   `yaml:"key"`
-	Operator string   `yaml:"operator"`
-	Values   []string `yaml:"values"`
-}
-
-type addressPool struct {
-	Protocol          Proto
-	Name              string
-	Addresses         []string
-	AvoidBuggyIPs     bool               `yaml:"avoid-buggy-ips"`
-	AutoAssign        *bool              `yaml:"auto-assign"`
-	BGPAdvertisements []bgpAdvertisement `yaml:"bgp-advertisements"`
-}
-
-type bgpAdvertisement struct {
-	AggregationLength   *int `yaml:"aggregation-length"`
-	AggregationLengthV6 *int `yaml:"aggregation-length-v6"`
-	LocalPref           *uint32
-	Communities         []string
-}
-
-type bfdProfile struct {
-	Name             string  `yaml:"name"`
-	ReceiveInterval  *uint32 `yaml:"receive-interval"`
-	TransmitInterval *uint32 `yaml:"transmit-interval"`
-	DetectMultiplier *uint32 `yaml:"detect-multiplier"`
-	EchoInterval     *uint32 `yaml:"echo-interval"`
-	EchoMode         bool    `yaml:"echo-mode"`
-	PassiveMode      bool    `yaml:"passive-mode"`
-	MinimumTTL       *uint32 `yaml:"minimum-ttl"`
+type CRSConfig struct {
```

Comment:
Maybe "CRsConfig?

Reply:
Yeah that name is pretty ugly, I knew it had to change while adding it :-)
K8sConfig or Resources or ClusterResources are valid alternatives, maybe the latter is the most convincing one (config.ClusterResources from outside the package)